src/mapmanager.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "imap.h"
+#include <map>
+#include <memory>
+#include <unordered_map>
+
+namespace bpftrace {
+
+/**
+   A container for all maps created during program execution.
+
+*/
+class MapManager
+{
+public:
+  MapManager(){};
+
+  MapManager(const MapManager &) = delete;
+  MapManager &operator=(const MapManager &) = delete;
+  MapManager(MapManager &&) = delete;
+  MapManager &operator=(MapManager &&) = delete;
+
+  /**
+     Store and lookup named maps
+  */
+  void Add(std::unique_ptr<IMap> map);
+  bool Has(const std::string &name);
+  IMap &Lookup(const std::string &name);
+  IMap &Lookup(ssize_t id);
+
+  IMap &operator[](ssize_t id)
+  {
+    return Lookup(id);
+  };
+  IMap &operator[](const std::string &name)
+  {
+    return Lookup(name);
+  };
+
+  /**
+     Iterate over named maps, vector like iteration
+  */
+  auto begin()
+  {
+    return maps_by_id_.begin();
+  };
+  auto end()
+  {
+    return maps_by_id_.end();
+  };
+
+  /**
+     Internal maps
+  */
+  enum class Type
```

Comment:
Not entirely sure about this part of it. Storing it as `elapsed_map_` etc like bpftrace did might be a better interface, with `std::optional<Map&> GetElapsed()` as getter?

Reply:
No strong opinion here. Seems easy enough to modify if we change our minds later.